### ISSUE TITLE
rclone move delete both files in local and remote of OneDrive Business

### ISSUE THREAD

**Issue by monologconnor** [rclone/rclone#5344]
#### What is the problem you are having with rclone?

I made the planned rclone task to transfer my file from the VPS to my OneDrive Business storage.

Mostly it works great except when transferring the large files, it seems like the file was been uploaded multiple times and been deleted by the rclone in both local and remote so I could find the files in OneDrive nowhere except the Recycle Bin with 3 copies

I tried to use the logging to find the problem, and the log says rclone could not match local and remote files with hash code then delete the file in remote, after that the rclone could not stop the local file to be deleted as well.

#### What is your rclone version (output from `rclone version`)
v1.51.0

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Ubuntu

#### Which cloud storage system are you using? (e.g. Google Drive)
Onedrive Business

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
rclone move localfolder remote:folder

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

2021/05/21 10:59:27 ERROR : Transferring.mp4: Failed to calculate src hash: hash: failed to stat: lstat Transferring.mp4: no such file or directory                       2021/05/21 10:59:27 ERROR : Transferring.mp4: corrupted on transfer: QuickXorHash hash differ "" vs "8b4b104e3517148a61c914fa02619968e901cb09"          2021/05/21 10:59:27 INFO  : Transferring.mp4: Removing failed copy                              
2021/05/21 10:59:27 ERROR : Transferring.mp4: Not deleting source as copy failed: corrupted on transfer: QuickXorHash hash differ "" vs "8b4b104e3517148a61c914fa02619968e901cb09"

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by animosity22** [comment]
That's a really old version. You should update to the latest and validate it still is a problem.

**Comment by monologconnor** [comment]
Oh okay, sorry I didn't notice that. I'll try to update in later days.
Just closing this issue first. If the same problem still happens I'll comment below.
Thanks!

**Comment by monologconnor** [comment]
Hi! I just updated to the latest version and tested recently, the situation of files being deleted after uploading to OneDrive Business still happened.

**Comment by ncw** [comment]
@monologconnor can you send a complete log from a simple example showing the problem with `-vv` please?

**Comment by monologconnor** [comment]
> @monologconnor can you send a complete log from a simple example showing the problem with `-vv` please?

Hi! Sorry for late replying.

I'll post the complete log here once I captured the error, since this error does not occur every time. 
btw, is it possible that the issue is caused by error reading the hash of the file since the file is being seeding in the Torrent Client? If it's possible, then I think it's my own problem, not related to rclone.

**Comment by monologconnor** [comment]
[output.log](https://github.com/rclone/rclone/files/6660456/output.log)
Hi! I think I just found where the problem comes after checking through the log.

In my situation, I have set an automatic plan to upload source files to remote every hour.
So when transferring a large file like size of 5gb, here will launch a rclone process A at very first and another process B after an hour while the process A was not completed.

When process A finished the transferring, it deleted the source file before exiting while process B is still going. And when process B was completed, the process could not open the source file for hash checking since the source file has been deleted already. Then the process B would destroy all transferring files in the remote due to failed hash check, which makes the file was deleted in both local and remote host.

I think I should set the automatic script which check if there is an on-going process before running rclone.

Thanks for ur help anyway! :)
